core/src/main/java/com/segment/analytics/kotlin/core/platform/Mediator.kt

```diff
@@ -27,29 +27,10 @@ internal class Mediator(internal val plugins: MutableList<Plugin>) {
             if (result != null) {
                 when (plugin) {
                     is DestinationPlugin -> {
-                        plugin.process(result)
-                    }
-                    is EventPlugin -> {
-                        when (result) {
-                            is IdentifyEvent -> {
-                                result = plugin.identify(result as IdentifyEvent)
-                            }
-                            is TrackEvent -> {
-                                result = plugin.track(result as TrackEvent)
-                            }
-                            is GroupEvent -> {
-                                result = plugin.group(result as GroupEvent)
-                            }
-                            is ScreenEvent -> {
-                                result = plugin.screen(result as ScreenEvent)
-                            }
-                            is AliasEvent -> {
-                                result = plugin.alias(result as AliasEvent)
-                            }
-                        }
+                        plugin.execute(result!!)
```

Comment:
i think this needs to be `result = plugin.execute(result)`

secondly, we should not be using `!!` since the result from one of the plugins could be null and this would then throw an NPE. lets wrap the call in a null check or use `?.let` and capture the state of result before using it as non-null

Reply:
hrm. `result = plugin.execute(result)` is a compile time error. we don't need a null check because we checked it on line 27

Reply:
actually scratch that first comment, didnt realize this was for DestinationPlugin